### ISSUE TITLE
NGINX continues serving an expired certificate

### ISSUE THREAD

**Issue by kontiainen** [home-assistant/addons#2870]
### Describe the issue you are experiencing

DuckDNS add-on had generated a new SSL certificate to replace an expired one, but NGINX did not recognize it and continued serving the expired one -> I was not able to get into my Home Assistant with iOS Companion app any more.

Restarting NGINX add-on fixed the situation. However, this feels like a bug - agreed? 

### What type of installation are you running?

Home Assistant OS

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

NGINX Home Assistant SSL proxy

### What is the version of the add-on?

3.2.0

### Steps to reproduce the issue

1. DuckDNS add-on renewed the certificate
2. Not possible to get into HA with companion app

(I don’t know how to reproduce this manually.)

### System Health information

”There are currently no repairs available”

**Comment by Dutch317** [comment]
Same setup on my install and had the same problem. It persisted through a HA reboot and only started working after restarting the addon.

**Comment by wolfwander** [comment]
I solved this monitoring the certificate through Certificate Integration e automating the restart of NGINX after certificate renewal. As I am behind a CG NAT , I need only the AAAA registry of my HASS gets registered and the DuckDNS add-on registers both A and AAAA , it doesn't suit my needs, so I register the DNS names using another automation.

**Comment by lmagyar** [comment]
I've run into this also. Without ssh or vpn I would have lost my connection to this device.

I've made a PR #2913 to fix this.

Until it is merged, I'm using my fork https://github.com/lmagyar/homeassistant-addon-nginx-proxy

**Comment by lmagyar** [comment]
I've checked the long time logs on host with `journalctl CONTAINER_NAME=addon_core_nginx_proxy`:
- My nginx add-on was restarted on Jan. 31, cert was renewed on Feb. 11, old cert expired on Mar. 13, and I lost connection to nginx endpoint on Mar. 13, and restarted nginx after that
- I haven't found any sign in nginx doc, that it reloads the new cert automatically
- I have found quite recent descriptions, on how to reload nginx to reload the new cert
- So even I'm just tinkering with it, I'm quite convinced nginx requires a reload to reload the renewed cert
